File: spoj/fashion_show.py
# http://www.spoj.com/problems/FASHION/

def read_participants():
    value = raw_input()
    return [int(i) for i in value.split(' ')]

def process():
    num_testcases = int(raw_input())
    count = 0
    while count < num_testcases:
        num_participants = int(raw_input())
        men = read_participants()
        women = read_participants()
        men.sort()
        women.sort()
        print sum([x * y for x, y in zip(men, women)])
        count += 1

process()

File: spoj/FashionShow.clj
;; http://www.spoj.com/problems/FASHION/

(ns fashion
  (:require [clojure.string :as str]))

(defn read_num[] 
    (def str1 (read-line))
    (def number (read-string str1))
    number
)

(defn read_participants[]
    (def nums (map read-string (str/split (read-line) #" ")))
    nums
)

(defn process_testcase[]
    (def num_participants (read_num))
    (def men (read_participants))
    (def women (read_participants))
    (def s (reduce + 0 (map * men women)))
    (println s)
)

(defn process[num_cases] 
    (if (= num_cases 0)
    true
     (do
        (process_testcase) 
        (process (- num_cases 1)) 
     )
    )
)

(def num_cases (read_num))
(process num_cases)